modules/data/src/dataservices/interfaces.ts

```diff
@@ -24,9 +31,28 @@ export interface RequestData {
 
 /**
  * A key/value map of parameters to be turned into an HTTP query string
- * Same as HttpClient's HttpParamsOptions which is NOT exported at package level
+ * Same as HttpClient's HttpParamsOptions which at the time of writing was
+ * NOT exported at package level
  * https://github.com/angular/angular/issues/22013
+ *
+ * @deprecated Use HttpOptions instead. getWithQuery still accepts QueryParams as its
+ * first argument, but HttpOptions.httpParams uses Angular's own HttpParamsOptions which
+ * HttpClient accepts as an argument.
  */
 export interface QueryParams {
-  [name: string]: string | string[];
+  [name: string]:
+    | string
+    | number
+    | boolean
+    | ReadonlyArray<string | number | boolean>;
+}
+
+/**
+ * Options that adhere to the constructor arguments for HttpParams and
+ * HttpHeaders.
+ */
+export interface HttpOptions {
+  httpParams?: HttpParamsOptions;
+  httpHeaders?: string | { [p: string]: string | string[] };
```

Comment:
can we use `HttpHeaders` here?

Reply:
reason was that I think `HttpHeaders` is not serializable - to be in the payload of an action it needs to be, correct?.  The HttpHeaders object is then created 'just in time' for the http request call.

However I now see that `HttpParamsOptions` isn't serializable either because of the `encoder?: HttpParameterCodec;` property.

I should probably really use a custom type for that too. Do you agree?

Reply:
Or could add doco that users shouldn't use the `encoder` property...

Reply:
I think introducing a serializable interface would be the best here.
We can convert it to queryparams/headers before sending the request.

Reply:
agree, doing now

Reply:
done